html_xblock/bleaching.py

```diff
@@ -0,0 +1,203 @@
+            'a',
+            'b',
+            'blockquote',
+            'code',
+            'em',
+            'h3',
+            'h4',
+            'h5',
+            'h6',
+            'i',
+            'img',
+            'li',
+            'span',
+            'strong',
+            'pre',
+            'ol',
+            'ul',
+            'p',
+            'pre'
+        ]
+
+        if not self.strict:
+            tags += ['h1', 'h2', 'script', 'sub', 'sup', 'div', 'abbr', 'iframe']
+
+        return tags
+
+    def _get_allowed_attributes(self):
+        """
+        This is an override to the original bleaching cleaner ALLOWED_ATTRIBUTES, it deals with two bleaching modes,
+        the strict mode, and the trusted mode.
+
+        :return: Allowed attributes depending on the bleaching mode
+        """
+        attributes = {
+            'a': ['href', 'title', 'target', 'rel'],
+            'img': ['src', 'alt', 'width', 'height'],
+            'p': ['style'],
+            'pre': ['class'],
+            'span': ['style'],
+            'ul': [],
+        }
+
+        if not self.strict:
+            attributes.update({'abbr': ['title']})
+            attributes['ul'].append('style')
+            attributes['img'].append('style')
+
+        return attributes
+
+    def _get_allowed_styles(self):
+        """
+        This is an override to the original bleaching cleaner ALLOWED_STYLES, it deals with two bleaching modes,
+        the strict mode, and the trusted mode.
+
+        :return: Allowed styles depending on the bleaching mode
+        """
+        styles = ['font-family', 'text-align', 'color', 'text-decoration', 'padding-left', 'padding-right']
+
+        if not self.strict:
+            styles += ['list-style-type', 'font-size', 'border-width', 'margin']
+
+        return styles
+
+    def _determine_values(self, other):
+        """
+        Return the values to be compared, if `other` is an instance of `str` then we will compare `other`'s value with
+        this instance's clean value. Else if `other` is an instance of this class we will compare the `other`'s
+        adulterate value (The original value) with the instance adulterate value as well.
+        :param other:
+        :return: A tuple of values to be compared.
+        """
+        if isinstance(other, str):
+            self_value = self.sanitized_value
+            other_value = other
+        elif isinstance(other, type(self)):
+            self_value = self.adulterated_value
+            other_value = other.adulterated_value
+        else:
+            raise TypeError('Unsupported operation between instances of \'{}\' and \'{}\''.format(
+                type(self).__name__, type(other).__name__))
+
+        return self_value, other_value
+
+    def __str__(self):
+        """
+        :return: The value of the text depending on the strictness level.
+        """
+        return self.sanitized_value if self.strict else self.adulterated_value
```

Comment:
Yes, but that was the idea from the beginning, right? We need to keep both versions and show only the `sanitized_value` on the user's end. We can make the `adulterated_value` not accessible once the object has been identified as a strict one, but I don't think that'd be useful in case we want later to see the original value especially when we want to display it in Studio for course managers.
I updated this to make it an object initialization time rather than at method execution time. Please tell me if you're thinking of something else or if I'm missing anything.